src/platform_impl/macos/observer.rs

```diff
@@ -152,31 +192,41 @@ impl RunLoop {
         flags: CFOptionFlags,
         priority: CFIndex,
         handler: CFRunLoopObserverCallBack,
+        context: *mut CFRunLoopObserverContext,
     ) {
         let observer = CFRunLoopObserverCreate(
             ptr::null_mut(),
             flags,
             ffi::TRUE, // Indicates we want this to run repeatedly
             priority,  // The lower the value, the sooner this will run
             handler,
-            ptr::null_mut(),
+            context,
         );
         CFRunLoopAddObserver(self.0, observer, kCFRunLoopCommonModes);
     }
 }
 
-pub fn setup_control_flow_observers() {
+pub(crate) fn setup_control_flow_observers(panic_info: Weak<PanicInfo>) {
```

Comment:
This function isn't visible outside of `platform_impl::macos` either way, right? I'm not sure if adding `(crate)` makes that more explicit or if it makes the overall module structure and their relative visibilities less clear.

(This comment applies to other things this PR marks `pub(crate)` too)

Reply:
As far as I understand, if these are just `pub` then they can still be reached from outside the crate. This also limits the optimizations that can be done when compiling the crate.

But I don't feel strongly about this, so I'm changing them to `pub`.